src/coreComponents/fileIO/Table/TableFormatter.cpp

```diff
@@ -0,0 +1,406 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file TableFormatter.cpp
+ */
+
+#include <numeric>
+#include "TableFormatter.hpp"
+namespace geos
+{
+
+TableFormatter::TableFormatter( TableLayout const & tableLayout ):
+  m_tableLayout( tableLayout )
+{}
+
+///////////////////////////////////////////////////////////////////////
+////// CSV Formatter implementation
+///////////////////////////////////////////////////////////////////////
+
+TableCSVFormatter::TableCSVFormatter( TableLayout const & tableLayout ):
+  TableFormatter( tableLayout )
+{
+  m_tableLayout = tableLayout;
+}
+
+string TableCSVFormatter::headerToString() const
+{
+  std::stringstream oss;
+  static constexpr string_view separator = ",";
+
+  for( std::size_t idxColumn = 0; idxColumn < m_tableLayout.getColumns().size(); ++idxColumn )
+  {
+    oss << m_tableLayout.getColumns()[idxColumn].m_parameter.columnName;
+    if( idxColumn < m_tableLayout.getColumns().size() - 1 )
+    {
+      oss << separator;
+    }
+  }
+  oss << "\n";
+  return oss.str();
+}
+
+string TableCSVFormatter::dataToString( TableData const & tableData ) const
+{
+  std::vector< std::vector< string > > const rowsValues = tableData.getTableDataRows();
+  std::ostringstream oss;
+
+  for( const auto & row : rowsValues )
+  {
+    for( size_t idxColumn = 0; idxColumn < row.size(); ++idxColumn )
+    {
+      oss << row[idxColumn];
+      if( idxColumn < row.size() - 1 )
+      {
+        oss << ",";
+      }
+    }
```

Comment:
```suggestion
oss << stringutilities::join( row.cbegin(), row.cend(), ",");
```
?

Reply:
Indeed I hadn't seen this utils function